src/mutation.cpp

```diff
@@ -12,98 +12,308 @@
 #include <algorithm> //std::min
 #include <sstream>
 
-std::vector<std::string> unpowered_traits;
-
-bool Character::has_trait(const std::string &b) const
+bool Character::has_mut(const muttype_id &flag) const
 {
-    // Look for active mutations and traits
-    return my_mutations.find( b ) != my_mutations.end();
+    return my_mutations.find(flag) != my_mutations.end();
 }
 
-bool Character::has_base_trait(const std::string &b) const
+bool Character::is_trait(const muttype_id &flag) const
 {
-    // Look only at base traits
-    return my_traits.find( b ) != my_traits.end();
+    // Returns 1 or 0 depending on the existence of the trait
```

Comment:
The comment is not really helpful. Does it return 0 if the trait is present or does it return 1 in that case? The function itself is defined as returning a boolean, why is this comment talking about integers? Why is the return statement returning an integer (actually as `size_t`) and not a boolean?

I know that works and how it works, but the comment is either pointless and can be removed, or should state explicitly what's the happening, like "returns whether the character has that trait" (and this should be in the comment of the function in the header). It also avoids all the ambiguity with "returns something depending on something else".